### DISCUSSION TITLE
CieLab measurements doesn't work

### DISCUSSION THREAD

**Discussion by SylvainArd** [ImageMagick/ImageMagick#5107] [Help]
hello, 
as I expected the measured CieLab values of the same photo of the color chart have nothing to do with the theoretical CieLab measurements!
What files do you need to attempt a debug?
Thank you
Best regards

**Comment by SylvainArd** [comment]
my QuantumRange is 255
my commands chain is :
scanin -dipn -G1.0 -v2 mire.tif ref\it8.cht R202202.txt
colprof -v -D"Scanner A" -qm -ax -ua mire
magick mire.tif +profile ICC -profile mire.icm -profile Lab-D50-Identity-elle-V4.icc -define quantum:format=floating-point -depth 32 out.tif
magick out.tif -crop 63x65+402+96 +repage ftxt: > cielab.txt
The files are :
https://www.dropbox.com/s/iib7grvvkzcistm/R202202.txt?dl=0
https://www.dropbox.com/s/6hc0fsene1lsnws/mire057.tif?dl=0
https://www.dropbox.com/s/r50l3aa12n8e075/mire.icm?dl=0

thank you for help

**Reply by fmw42** [reply]
You are going to print 63x65=4095 pixel values.  Is that what you want?

**Comment by snibgo** [comment]
What version of IM do you use? Please show the text output from "magick -version". If your version is not HDRI, your process won't work correctly, because your process needs negative values.

**Comment by SylvainArd** [comment]
yes I do a medium on these values
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers


Le dim. 1 mai 2022 à 20:11, Fred Weinhaus ***@***.***> a
écrit :

**Comment by SylvainArd** [comment]
I have fixed this bug by clean up precedent installation of IM and restart
my computer
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
I have installed the HDRI Q16 version, the result is the same, only the L
value is good, A and B are bad, I think it is the calculus which is bad
for L I do : 100*L/QuantumRange
for A and B I do : 255*a/QuantumRange-128, what is bad ?
thank you

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
Why are you subtracting 128? Don't.

**Comment by SylvainArd** [comment]
I want to get values from -128 to 127
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
the E22 theorical values are : 37.27  -23.65   26.36
what I get is : 24415,58981.5,7324.5
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
no it doesn't works, there are many patches where it doesn't works, this is
strange because L is good !
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
ah si c'est bon (sorry I haven't the translation), I have not read the
correct line
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
I found a calculus which works on all patches with a delta of 5 :
a:=255*a/QuantumRange;
    if a>127 then
      a:=a-255;
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
C:\Windows\System32>magick -version
Version: ImageMagick 7.1.0-30 Q8 x64 c8ecfc4:20220416
https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl
lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193131105)

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
the L value is good approximatively for all patches
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
I installed HDRI version, now if I do the command inside the cmd I got
out.tif but if I launch it by CreateProcess I have the following error :
C:\Users\Sylvain ARD\Dropbox\Documents
Sylvain\autoentrepreneur\Contrats\Projet
couleurs\PROGRAMME\Delphi\Win32\Debug>magick in.tif +profile ICC -profile
"C:\Users\Sylvain ARD\Dropbox\Documents
Sylvain\autoentrepreneur\Contrats\Projet couleurs\scanner.icm" -profile
Lab-D50-Identity-elle-V4.icc  out.tif  1>res.txt
magick: UnableToOpenConfigureFile `delegates.xml' @
warning/configure.c/GetConfigureOptions/705.
magick: RegistryKeyLookupFailed `CoderModulesPath' @
error/module.c/GetMagickModulePath/665.
magick: NoDecodeDelegateForThisImageFormat `TIFF' @
error/constitute.c/ReadImage/737.

please help me !

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> magick: UnableToOpenConfigureFile `delegates.xml' 

This suggests something is wrong with your installation. I suggest you remove all copies of IM, and re-install it.

**Comment by SylvainArd** [comment]
I have fixed this bug before, thank you
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers
